src/main/java/picard/analysis/CompareMetrics.java

```diff
@@ -50,35 +70,244 @@
+        if (METRIC_ALLOWABLE_RELATIVE_CHANGE != null) {
+            for (String diffs : METRIC_ALLOWABLE_RELATIVE_CHANGE) {
+                String[] pair = diffs.split(":");
+                if (pair.length == 2) {
+                    String name = pair[0];
+                    try {
+                        double value = Double.parseDouble(pair[1]);
+                        if (value > 0) {
+                            MetricToAllowableRelativeChange.put(name, value);
+                        } else {
+                            errs.add("Value for numeric component of Argument 'METRIC_ALLOWABLE_RELATIVE_CHANGE' must be > 0.0");
+                        }
+                    } catch (NumberFormatException ne) {
+                        errs.add("Invalid value for numeric component of Argument 'METRIC_ALLOWABLE_RELATIVE_CHANGE'");
+                    }
+                } else {
+                    errs.add("Invalid value for Argument 'METRIC_ALLOWABLE_RELATIVE_CHANGE'");
+                }
+            }
+        }
+
+        if (errs.isEmpty()) {
+            return null;
+        } else {
+            return errs.toArray(new String[0]);
+        }
+    }
+
+    private int compareMetricsFiles(final File metricFile1, final File metricFile2) throws IOException, IllegalAccessException {
+        final MetricsFile<?, ?> mf1 = new MetricsFile<>();
+        final MetricsFile<?, ?> mf2 = new MetricsFile<>();
+        mf1.read(new FileReader(metricFile1));
+        mf2.read(new FileReader(metricFile2));
+        if (mf1.areMetricsEqual(mf2)) {
+            return 0;
+        } else if (!mf1.getMetrics().get(0).getClass().equals(mf2.getMetrics().get(0).getClass())) {
+            throw new PicardException("Metrics are of differing class between " + metricFile1.getAbsolutePath() + " and " + metricFile2.getAbsolutePath());
+        } else if (!mf1.getMetricsColumnLabels().equals(mf2.getMetricsColumnLabels())) {
+            log.error("Metric columns differ between " + metricFile1.getAbsolutePath() + " and " + metricFile2.getAbsolutePath());
+            return 1;
+        } else if (mf1.getMetrics().size() != mf2.getMetrics().size()) {
+            log.error("Number of metric rows differ between " + metricFile1.getAbsolutePath() + " and " + metricFile2.getAbsolutePath());
+            return 1;
+        }
+
+        validateMetricNames(mf1, metricFile1, METRICS_TO_IGNORE);
+        validateMetricNames(mf1, metricFile1, MetricToAllowableRelativeChange.keySet());
+
+        Set<String> metricsToIgnore = new HashSet<>(METRICS_TO_IGNORE);
+
+        final Class<? extends MetricBase> metricClass = mf1.getMetrics().get(0).getClass();
+
+        int retVal = 0;
+        int rowNumber = -1;
+        final Field[] fields = metricClass.getFields();
+        Iterator<?> mf1Iterator = mf1.getMetrics().iterator();
+        Iterator<?> mf2Iterator = mf2.getMetrics().iterator();
+        while (mf1Iterator.hasNext() && mf2Iterator.hasNext()) {
```

Comment:
It doesn't hurt to check both, but since the code already asserts that the sizes are equal, they should only ever both be true. So the code only needs to check one here.

Reply:
True that.